packages/melange.ppx/ppx.ml

```diff
@@ -256,8 +370,32 @@ let transform_external pval_name pval_attributes pval_loc pval_type =
       (* When mel.send.pipe is used, it's treated as a funcion *)
       if Option.is_some (get_send_pipe pval_attributes) then
         transform_external_arrow ~loc pval_name pval_attributes pval_type
-      else if has_mel_module_attr pval_attributes then
-        [%stri [%%ocaml.error [%e mel_module_found_in_native_message ~loc]]]
+      else if Mel_module.has_attr pval_attributes then
+        match Mel_module.asset_payload pval_attributes with
+        | None ->
+            (* If it doesn't have asset payload, we error out as it must be some .js module or package being imported *)
+            [%stri [%%ocaml.error [%e mel_module_found_in_native_message ~loc]]]
+        | Some str ->
+            (* If it has asset payload (file with extension), calculate hash and replace external *)
+            let name = Builder.pvar ~loc:pval_name.loc pval_name.txt in
+            let path =
+              let asset_path = Filename.(concat (dirname module_path) str) in
+              (* todo: maybe read line by line of buffered for large files *)
+              let ic = open_in asset_path in
+              let n = in_channel_length ic in
+              let s = really_input_string ic n in
+              close_in ic;
```

Comment:
I used `In_channel.with_open_bin file In_channel.input_all` in the past, if you want to avoid having to open/close_in